File: Hook/src/main/java/www/leslie/org/hook/MainActivity.java
package www.leslie.org.hook;

import android.app.Activity;
import android.app.Instrumentation;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.Scroller;

import java.lang.reflect.Field;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    public MyButton btn;
    public Scroller scroller;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
//        startService(new Intent(this,TargetService.class));
       // startActivity(new Intent(this,HookActivity.class));
       // test();

//        replaceActivityInstrumentation();
        btn = findViewById(R.id.btn);


    }

    private void test() {
        for(int i=0;i<1000;i++){
            System.out.println(i);
        }

    }
    public void click(View view){
       // test();
        boolean b = view instanceof Button;
        System.out.println(b);
        System.out.println(view.getScrollX());
        btn.smoothScrollTo(-400,0);
        System.out.println(view.getScrollX());



    }


    private void replaceActivityInstrumentation() {
        Field field = null;
        try {
            field = Activity.class.getDeclaredField("mInstrumentation");
            field.setAccessible(true);
            Instrumentation instrumentation = (Instrumentation) field.get((Activity) this);
            InstrumentationPrxy prxy = new InstrumentationPrxy(instrumentation);

            field.set((Activity)this,prxy);

        } catch (NoSuchFieldException e) {
            e.printStackTrace();
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        }

    }
}

File: kotlinTest/src/main/java/www/leslie/org/kotlintest/MainActivity.kt
package www.leslie.org.kotlintest

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

import android.widget.TextView

class MainActivity : AppCompatActivity() {
    //1 2

    private var a:Int=10;
    private var c:Int=5

    public fun  say(i: Int):Boolean {
        //1 2 3
        try {
            val ha = TextView(this);
            var c:String?=null
            if (c == null) {

            }
            var s:String?=null
        } catch (e: Exception) {
        }
        return true;

    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)




    }
}
